fix(lsp): improve err handling when loading source text

camc314 opened the pull request:
note i didn't test this, but it's panicing because `fs::read_to_string` is returning an error as the path no longer exists.

fixes https://github.com/oxc-project/oxc/issues/8023

camc314 commented:
* **#8025** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8025?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8025?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8025).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Sysix commented:
Thanks 👍 
I am still asking myself why vscode has sent `textDocument/didOpen`.  But this should be another problem ;)

camc314 commented:
> Thanks 👍 I am still asking myself why vscode has sent `textDocument/didOpen`. But this should be another problem ;)

i think the `textDocument/didOpen` that it sends is correct - it's just confusing for us to handle because it's opened a buffer that references a deleted file.

camc314 commented:
<img width="855" alt="Screenshot 2024-12-19 at 19 06 06" src="https://github.com/user-attachments/assets/b3d646ce-553d-49b1-89ab-ddcef6484ab8" />

camc314 commented:
### Merge activity

* **Dec 19, 2:07 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 19, 2:07 PM EST**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

camc314 committed:
fix(lsp): improve err handling when loading source text (#8025)

note i didn't test this, but it's panicing because `fs::read_to_string` is returning an error as the path no longer exists.

fixes https://github.com/oxc-project/oxc/issues/8023
